doc/user/layouts/shortcodes/json-parser.html

```diff
@@ -73,43 +82,65 @@
 /// Recursively iterates through the provided object, tracking the chain
 /// of parent fields for later use in naming and desctructuring.
 function expandObject(object, parents, columns) {
```

Comment:
The `expandObject` func refactor is necessary to fix the bug reported in: https://github.com/MaterializeInc/materialize/issues/25178

Mainly adding some extra logic to handle nested arrays and empty arrays.

Took the time to split this into separate functions for simplicity, but totally open to feedback.